File: frontend/src/types/agent.ts
/** Represents an AI agent that can be assigned to work items. */
export interface Agent {
  id: number
  workspace_id: number
  name: string
  avatar: string                           // emoji, e.g. "🤖"
  agent_type: 'builtin' | 'custom'
  capabilities: string[]                   // JSON array of tool names
  status: 'active' | 'inactive'
  model_override?: string
  system_prompt?: string
  created_by_id: number
  created_at: string
  updated_at: string
}

/** Records every action an AI agent performs — used for audit trail. */
export interface AgentActivity {
  id: number
  agent_id: number
  issue_id?: number
  action: string                           // "dispatch" | "auto_triage" | "auto_assign" | "mention" | "summarize" | "custom"
  result_summary: string
  executed_at: string
  agent_name: string
  task_context?: string
  rating?: number | null                     // 1=positive, -1=negative, null=no feedback
  created_at: string
}

/** Request body for dispatching a task to an agent. */
export interface AgentDispatchRequest {
  task: string
  issue_id?: number
  project_id?: number
}

/** Request body for creating a new agent. */
export interface AgentCreateRequest {
  name: string
  avatar?: string
  agent_type?: 'builtin' | 'custom'
  capabilities?: string[]
  status?: 'active' | 'inactive'
  model_override?: string
  system_prompt?: string
}

/** Request body for updating an existing agent. */
export interface AgentUpdateRequest {
  name?: string
  avatar?: string
  agent_type?: 'builtin' | 'custom'
  capabilities?: string[]
  status?: 'active' | 'inactive'
  model_override?: string
  system_prompt?: string
}

File: agent-service/internal/model/agent.go
package model

import (
	"encoding/json"
	"time"
)

// Agent represents an AI agent that can be assigned to work items and @mentioned in comments.
type Agent struct {
	ID           uint64          `gorm:"primaryKey" json:"id"`
	CreatedAt    time.Time       `json:"created_at"`
	UpdatedAt    time.Time       `json:"updated_at"`
	DeletedAt    *time.Time      `gorm:"index" json:"-"`
	CreatedByID  *uint64         `json:"created_by_id"`
	UpdatedByID  *uint64         `json:"updated_by_id"`

	WorkspaceID   uint64          `gorm:"not null;index" json:"workspace_id"`
	Name          string          `gorm:"size:128;not null" json:"name"`
	Avatar        string          `gorm:"size:10;default:🤖" json:"avatar"`
	AgentType     string          `gorm:"size:20;default:builtin" json:"agent_type"`
	Capabilities  json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"capabilities"`
	Status        string          `gorm:"size:20;default:active" json:"status"`
	ModelOverride *string         `gorm:"size:50" json:"model_override,omitempty"`
	SystemPrompt  *string         `gorm:"type:text" json:"system_prompt,omitempty"`
}

func (Agent) TableName() string { return "agents" }

// AgentActivity records every action an AI agent performs for audit trail purposes.
type AgentActivity struct {
	ID           uint64     `gorm:"primaryKey" json:"id"`
	CreatedAt    time.Time  `json:"created_at"`
	UpdatedAt    time.Time  `json:"updated_at"`
	DeletedAt    *time.Time `gorm:"index" json:"-"`
	CreatedByID  *uint64    `json:"created_by_id"`
	UpdatedByID  *uint64    `json:"updated_by_id"`

	AgentID       uint64    `gorm:"not null;index" json:"agent_id"`
	IssueID       *uint64   `gorm:"index" json:"issue_id"`
	Action        string    `gorm:"size:50;not null" json:"action"`
	ResultSummary string    `gorm:"type:text" json:"result_summary"`
	Rating        *int      `gorm:"default:null" json:"rating,omitempty"`
	ExecutedAt    time.Time `gorm:"autoCreateTime" json:"executed_at"`
	AgentName     string    `gorm:"size:128" json:"agent_name"`
	TaskContext   *string   `gorm:"type:text" json:"task_context,omitempty"`
}

func (AgentActivity) TableName() string { return "agent_activities" }
